src/easydiffraction/job.py

```diff
@@ -425,16 +425,18 @@ def add_experiment_from_file(self, file_url: str) -> None:
 
         self.update_job_type()
         # re-do the sample in case of type change.
-        if self.sample.parameters.name != self.experiment.parameters.name:
+        #if self.sample.parameters.name != self.experiment.parameters.name:
             # Different type read in (likely TOF), so re-create the sample
-            parameters = self.experiment.parameters
-            pattern = self.experiment.pattern
-            phases = self.sample.phases
-            name = self.sample.name
-            self.sample = Sample(name, parameters=parameters, pattern=pattern, phases=phases)
-            self.sample.parameters = self.experiment.parameters
-            self.update_job_type()
-            self.update_interface()
+        # re-do the sample always, as parameter values can change in the experiment.parameters compared to
+        # the sample.parameters (e.g. dtt1 read from CIF in Scipp format)
+        parameters = self.experiment.parameters
+        pattern = self.experiment.pattern
+        phases = self.sample.phases
+        name = self.sample.name
+        self.sample = Sample(name, parameters=parameters, pattern=pattern, phases=phases)
+        self.sample.parameters = self.experiment.parameters
+        self.update_job_type()
+        self.update_interface()
```

Comment:
Are you sure? We would only need to recreate the sample when the job type changes, otherwise sample params are directly linked to experiment params.

Reply:
Running `tests/integration_tests/test_scipp/test_scipp.py` without this results in the job type not being changed, but the parameters `dtt1` and `dtt2` are read from CIF and updated in `self.experiment.parameters`, but not in `self.sample.parameters`. Then accessing them in the standard way via the job, `job.parameters.dtt1`, still shows the old result and test fails. So for the moment I decided to recreate the sample every time. Anyway, this part should be thoroughly refactored in the near future. But if you see a simple fix that doesn't require recreating the sample, I will of course add it.